test/strings.jl

```diff
@@ -1423,6 +1423,19 @@ for T in [Int8, Int16, Int32, Int64, Int128]
     end
 end
 
+# issue #11142
+s = "abcdefghij"
+sp = pointer(s)
+@test ascii(sp) == s
+@test ascii(sp,5) == "abcde"
+@test typeof(ascii(sp)) == ASCIIString
+@test typeof(utf8(sp)) == UTF8String
+s = "abcde\uff\u2000\U1f596"
+sp = pointer(s)
+@test utf8(sp) == s
+@test utf8(sp,5) == "abcde"
+@test typeof(utf8(sp)) == UTF8String
+
```

Comment:
Seems like there should also be a `@test_throws` if you try to convert the second `sp` into `ascii`.